runelite-client/src/main/java/net/runelite/client/plugins/tileindicators/TileIndicatorsOverlay.java

```diff
@@ -106,6 +106,9 @@ private void renderTile(final Graphics2D graphics, final LocalPoint dest, final
 			return;
 		}
 
+		graphics.setColor(new Color(fillColor.getRed(), fillColor.getGreen(), fillColor.getBlue(), fillColor.polyAlpha()));
+		graphics.fillPolygon(poly);
```

Comment:
`renderPolygon` is already doing a `fill` with the polygon with `new Color(0, 0, 0, 50)` - you should not be drawing over it again but instead call the overloaded `renderPolygon` which accepts a fill color.